src/graphql.rs

```diff
@@ -542,42 +538,24 @@ where
     A: ObjectType,
     NodesField: ConnectionNameType,
 {
-    match (
-        first.is_some(),
-        last.is_some(),
-        before.is_some(),
-        after.is_some(),
-    ) {
-        (true, true, _, _) => return Err("cannot provide both `first` and `last`".into()),
-        (_, _, true, true) => return Err("cannot provide both `before` and `after`".into()),
-        (true, _, true, _) => return Err("cannot provide both `first` and `before`".into()),
-        (_, true, _, true) => return Err("cannot provide both `last` and `after`".into()),
-        (false, false, false, _) => {
-            first = Some(DEFAULT_CONNECTION_SIZE);
-        }
-        (false, false, true, false) => {
-            last = Some(DEFAULT_CONNECTION_SIZE);
-        }
-        _ => {}
-    }
-
-    let after = if let Some(after) = after {
-        Some(decode_cursor(&after).ok_or("invalid cursor `after`")?)
-    } else {
-        None
-    };
-    let before = if let Some(before) = before {
-        Some(decode_cursor(&before).ok_or("invalid cursor `before`")?)
-    } else {
-        None
-    };
-
     let (nodes, has_previous, has_next) = if let Some(first) = first {
-        let (nodes, has_more) = collect_edges(table, Direction::Forward, after, before, first);
+        let (nodes, has_more) = collect_edges(
+            table,
+            Direction::Forward,
+            after,
+            before,
+            first.to_usize().unwrap_or_default(),
```

Comment:
`first` is `usize` in the first place. Why does it need to be converted?

Reply:
Before one of the commits, `first` and `last` used to be `i32` but after it changed to `usize` , this was no longer needed. I thought I included it in my last commit, but I found another case where type conversion is no longer needed and included it in that commit which I rebased